src/core_plugins/console/public/src/input.js

```diff
@@ -13,232 +13,238 @@ let es = require('./es');
 let history = require('./history');
 import uiModules from 'ui/modules';
 
-var $el = $('#editor');
-var input = new SenseEditor($el);
-uiModules.get('app/sense').setupResizeCheckerForRootEditors($el, input, output);
+let cached;
+export default function getInput($el, $actionsEl, $copyAsCurlEl, output) {
+  if (!$el) return cached;
 
-input.autocomplete = new Autocomplete(input);
+  // var $el = $('#editor');
+  var input = new SenseEditor($el);
 
-input.$actions = $("#editor_actions");
+  uiModules.get('app/sense').setupResizeCheckerForRootEditors($el, input, output);
 
-input.commands.addCommand({
-  name: 'auto indent request',
-  bindKey: {win: 'Ctrl-I', mac: 'Command-I'},
-  exec: function () {
-    input.autoIndent();
-  }
-});
-input.commands.addCommand({
-  name: 'move to previous request start or end',
-  bindKey: {win: 'Ctrl-Up', mac: 'Command-Up'},
-  exec: function () {
-    input.moveToPreviousRequestEdge()
-  }
-});
-input.commands.addCommand({
-  name: 'move to next request start or end',
-  bindKey: {win: 'Ctrl-Down', mac: 'Command-Down'},
-  exec: function () {
-    input.moveToNextRequestEdge()
-  }
-});
-
-
-/**
- * COPY AS CURL
- *
- * Since the copy functionality is powered by a flash movie (via ZeroClipboard)
- * the only way to trigger the copy is with a litteral mouseclick from the user.
- *
- * The original shortcut will now just open the menu and highlight the
- *
- */
-var $copyAsCURL = $('#copy_as_curl');
-var zc = (function setupZeroClipboard() {
-  var zc = new ZeroClipboard($copyAsCURL); // the ZeroClipboard instance
-
-  zc.on('wrongflash noflash', function () {
-    if (!storage.get('flash_warning_shown')) {
-      alert('Console needs flash version 10.0 or greater in order to provide "Copy as cURL" functionality');
-      storage.set('flash_warning_shown', 'true');
+  input.autocomplete = new Autocomplete(input);
+
+  input.$actions = $actionsEl; // $("#editor_actions");
```

Comment:
Oh wait, is this commented-out code, or supposed to be a helpful comment? If the latter, I don't think it's very helpful since it's so brittle and can easily become misleading if we change the ID of the element we're passing in.